test/cppsuite/test_harness/workload/workload_validation.h

```diff
@@ -48,51 +48,35 @@ class workload_validation {
      * Validate the on disk data against what has been tracked during the test.
      * - The first step is to replay the tracked operations so a representation in memory of the
      * collections is created. This representation is then compared to what is on disk.
-     * - The second step is to go through what has been saved on disk and make sure the memory
-     * representation has the same data.
-     * operation_table_name is the collection that contains all the operations about the key/value
-     * pairs in the different collections used during the test. schema_table_name is the collection
-     * that contains all the operations about the creation or deletion of collections during the
-     * test.
+     * operation_table_name: collection that contains all the operations about the key/value
+     * pairs in the different collections used during the test.
+     * schema_table_name: collection that contains all the operations about the creation or deletion
+     * of collections during the test.
      */
     bool
     validate(const std::string &operation_table_name, const std::string &schema_table_name,
       database &database)
     {
+        WT_CURSOR *cursor;
         WT_SESSION *session;
-        std::string collection_name;
-        /* Existing collections after the test. */
+        wt_timestamp_t key_timestamp;
         std::vector<std::string> created_collections, deleted_collections;
+        const char *key, *key_collection_name, *value;
+        int value_operation_type;
+        std::string current_collection_name;
         bool is_valid = true;
 
         session = connection_manager::instance().create_session();
 
         /* Retrieve the collections that were created and deleted during the test. */
-        collection_name = schema_table_name;
         parse_schema_tracking_table(
-          session, collection_name, created_collections, deleted_collections);
-
-        /* Make sure they exist in memory. */
-        for (auto const &it : created_collections) {
-            if (database.collections.count(it) == 0) {
-                debug_print("Collection missing in memory: " + it, DEBUG_ERROR);
-                is_valid = false;
-                break;
-            }
-        }
-
-        if (!is_valid)
-            return (is_valid);
+          session, schema_table_name, created_collections, deleted_collections);
 
-        /* Make sure they don't exist in memory nor on disk. */
+        /*
+         * Make sure the deleted collections do not exist on disk. The created collections are
+         * checked in check_reference.
+         */
         for (auto const &it : deleted_collections) {
-            if (database.collections.count(it) > 0) {
-                debug_print(
-                  "Collection present in memory while it has been tracked as deleted: " + it,
-                  DEBUG_ERROR);
-                is_valid = false;
```

Comment:
Rather than keeping is_valid, as soon as we detect an error we should just testutil_die.